src/ApparentHorizons/StrahlkorperCoordsInDifferentFrame.cpp

```diff
@@ -55,38 +55,85 @@ void strahlkorper_coords_in_different_frame(
   // Block by Block.
   for (const auto& block : domain.blocks()) {
-    // Once there are more possible source frames than the grid frame
-    // (i.e. the distorted frame), then this static_assert will change,
-    // and there will be an `if constexpr` below to treat the different
-    // possible source frames.
-    static_assert(std::is_same_v<SrcFrame, ::Frame::Grid>,
-                  "Source frame must currently be Grid frame");
     static_assert(std::is_same_v<DestFrame, ::Frame::Inertial>,
                   "Destination frame must currently be Inertial frame");
-    const auto& grid_to_inertial_map = block.moving_mesh_grid_to_inertial_map();
-    const auto& logical_to_grid_map = block.moving_mesh_logical_to_grid_map();
-    // Fill only those dest_cartesian_coords that are in this Block.
-    // Determine which coords are in this Block by checking if the
-    // inverse grid-to-logical map yields logical coords between -1 and 1.
-    for (size_t s = 0; s < get<0>(src_cartesian_coords).size(); ++s) {
-      const tnsr::I<double, 3, SrcFrame> x_src{
-          {get<0>(src_cartesian_coords)[s], get<1>(src_cartesian_coords)[s],
-           get<2>(src_cartesian_coords)[s]}};
-      const auto x_logical = logical_to_grid_map.inverse(x_src);
-      // x_logical might be an empty std::optional.
-      if (x_logical.has_value() and get<0>(x_logical.value()) <= 1.0 and
-          get<0>(x_logical.value()) >= -1.0 and
-          get<1>(x_logical.value()) <= 1.0 and
-          get<1>(x_logical.value()) >= -1.0 and
-          get<2>(x_logical.value()) <= 1.0 and
-          get<2>(x_logical.value()) >= -1.0) {
-        const auto x_dest =
-            grid_to_inertial_map(x_src, time, functions_of_time);
-        get<0>(*dest_cartesian_coords)[s] = get<0>(x_dest);
-        get<1>(*dest_cartesian_coords)[s] = get<1>(x_dest);
-        get<2>(*dest_cartesian_coords)[s] = get<2>(x_dest);
-        // Note that if a point is on the boundary of two or more
-        // Blocks, it might get filled twice, but that's ok.
+    if constexpr (std::is_same_v<SrcFrame, ::Frame::Grid>) {
+      const auto& grid_to_inertial_map =
+          block.moving_mesh_grid_to_inertial_map();
+      const auto& logical_to_grid_map = block.moving_mesh_logical_to_grid_map();
+      // Fill only those dest_cartesian_coords that are in this Block.
+      // Determine which coords are in this Block by checking if the
+      // inverse grid-to-logical map yields logical coords between -1 and 1.
+      for (size_t s = 0; s < get<0>(src_cartesian_coords).size(); ++s) {
+        const tnsr::I<double, 3, SrcFrame> x_src{
+            {get<0>(src_cartesian_coords)[s], get<1>(src_cartesian_coords)[s],
+             get<2>(src_cartesian_coords)[s]}};
+        const auto x_logical = logical_to_grid_map.inverse(x_src);
+        const double one_plus_eps =
+            1.0 + std::numeric_limits<double>::epsilon() * 100.0;
```

Comment:
Suggestion: to avoid potential bugs like this, call `block_logical_coordinates` before the loop over blocks. It will do the SrcFrame -> BlockLogical mapping and return a vector of optionals, one for each grid point. Then, in this loop over grid points, check `block_logical_coords[s].has_value() and block_logical_coords[s]->id == block.id()`.

Reply:
Even better: if I call `block_logical_coordinates`, then I think I don't need to loop over blocks at all here.  I just need to loop over points and call either `grid_to_inertial_map` or `grid_to_distorted_map` for the (guaranteed unique) Block that is returned by `block_logical_coordinates` for that point.  Any objection to me doing that?

Reply:
sounds good

Reply:
Made this change and squashed (and rebased).